drafting the intro lesson (#2)
UW-Madison-DataScience/R-gitlab-lesson: episodes/intro.Rmd

---
title: "Introduction to Version Control and GitLab"
teaching: 10
exercises: 2

> based on: Software Carpentry Git-Novice Lesson

Reviewer: ```suggestion
```
Removing and adding text below

---

:::::::::::::::::::::::::::::::::::::: questions 

- What is Version Control?
- What is the difference between Git, GitHub, and GitLab?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Understand the utility and benefits of automated version control systems.
- Differentiate between Git, GitHub, and GitLab.
- Explain why GitLab was chosen over other platforms like GitHub.

::::::::::::::::::::::::::::::::::::::::::::::::

> We'll start by exploring how version control can be used to keep track of what one person did and when. Even if you aren't collaborating with other people,

Reviewer: ```suggestion
*This lesson is based on [Software Carpentry's Git Novice Lesson](https://swcarpentry.github.io/git-novice/)* and images are reused from that lesson.

We'll start by exploring how version control can be used to keep track of what one person did and when. Even if you aren't collaborating with other people,
```

:::::::::::::::  solution

## Solution

- Recovering the excellent version is only possible if you created a copy of the old version of the paper. The danger of losing good versions often leads to the problematic workflow illustrated in the PhD Comics cartoon at the top of this page.

- Collaborative writing with traditional word processors is cumbersome. Either every collaborator has to work on a document sequentially (slowing down the process of writing), or you have to send out a version to all collaborators and manually merge their comments into your document. The 'track changes' or 'record changes' option can highlight changes for you and simplifies merging, but as soon as you accept changes you will lose their history. You will then no longer know who suggested that change, why it was suggested, or when it was merged into the rest of the document. Even online word processors like Google Docs or Microsoft Office Online do not fully resolve these problems.   

::::::::::::::::::::::::: 

## Version Control Platforms

Today we will explore using the GitLab with RStudio, but how is GitLab different from Git and GitHub?

> Git is an open-source tool that tracks your changes. GitHub and GitLab are repositories that collect the complete history of those changes. GitHub offers free and paid plans, and as a member of UW-Madison, you have access to GitLab, which provides additional features such as licensing protections, authentication and identity management through your NetID, and support from DoIT Shared Tools. 

Reviewer: ```suggestion
Git is an open-source tool that tracks your changes. GitHub and GitLab are hosts on the web that collect the complete history of those changes. GitHub offers free and paid plans, and as a member of UW-Madison, you have access to GitLab, which provides additional features such as licensing protections, authentication and identity management through your NetID, and support from DoIT Shared Tools. 
```
